build-system/tasks/dist.js

```diff
@@ -276,6 +277,38 @@ function copyParsers() {
   });
 }
 
+/**
+ * Obtain a recursive file listing of a directory
+ * @param {string} dest - Directory to be scanned
+ * @return {Array} - All files found in directory
+ */
+async function walk(dest) {
+  const filelist = [];
+  const files = await fs.readdir(dest);
+
+  for (let i = 0; i < files.length; i++) {
+    const file = `${dest}/${files[i]}`;
+
+    fs.statSync(file).isDirectory()
+      ? Array.prototype.push.apply(filelist, await walk(file))
+      : filelist.push(file);
+  }
+
+  return filelist;
+}
+
+/**
+ * Generate a listing of all files in dist/ and save as dist/files.txt
+ */
+async function generateFileListing() {
+  const distDir = 'dist';
+  const filesOut = `${distDir}/files.txt`;
+  fs.writeFileSync(filesOut, '');
+  const files = (await walk(distDir)).map(f => f.replace(`${distDir}/`, ''));
+  fs.writeFileSync(filesOut, files.join('\n'));
+  log('Generated', cyan(filesOut));
```

Comment:
To keep the logging consistent with other parts of `dist`:
1. Add this line at the top of the function:
```
const startTime = Date.now();
```

2. Replace this log statement with:
```
endBuildStep('Generated', filesOut, startTime);
```

Reply:
Since this is not a parallel build task, the timekeeping does not pick up from where other build tasks finish. So the time output always reads as only a few milliseconds. Is this something you still want?

Reply:
Using `endBuildStep` is desirable because it automatically supports less verbose logging mode (used during CI). No worries, re: the short duration of the task.